File: RobotLib/src/main/java/com/acmerobotics/library/cameraoverlay/message/RequestFrameCommand.java
package com.acmerobotics.library.cameraoverlay.message;

import android.graphics.Bitmap;

public class RequestFrameCommand extends Command<Bitmap> {
    public RequestFrameCommand() {
        super(MessageType.REQUEST_FRAME);
    }

    @Override
    public Class<? extends Response<Bitmap>> getResponseClass() {
        return ReceiveFrameResponse.class;
    }

    @Override
    public byte[] payloadToByteArray() {
        return new byte[0];
    }

    @Override
    public void payloadFromByteArray(byte[] bytes) {

    }
}

File: CameraOverlay/src/main/java/com/acmerobotics/cameraoverlay/message/RequestFrameCommand.kt
package com.acmerobotics.cameraoverlay.message

import android.graphics.Bitmap

class RequestFrameCommand: Command<Bitmap>(MessageType.REQUEST_FRAME) {
    override fun getResponseType(): Class<out Response<Bitmap>> {
        return ReceiveFrameResponse::class.java
    }

    override fun payloadToByteArray(): ByteArray = ByteArray(0)
    override fun payloadFromByteArray(byteArray: ByteArray) {}

}
